src/Layouts/HeaderBar.vala

```diff
@@ -164,17 +118,181 @@ public class Akira.Layouts.HeaderBar : Gtk.HeaderBar {
+        var grid = new Gtk.Grid ();
+        grid.margin_bottom = 3;
+        grid.orientation = Gtk.Orientation.VERTICAL;
+        grid.width_request = 200;
+        grid.name = "main";
+
+        var artboard = new Akira.Partials.PopoverButton (_("Artboard"), "window-new-symbolic", {"A"});
+
+        var separator = new Gtk.Separator (Gtk.Orientation.HORIZONTAL);
+        separator.margin_top = separator.margin_bottom = 3;
+
+        // Create the shapes submenu
+        var shapes_grid = new Gtk.Grid ();
+        shapes_grid.margin_bottom = 3;
+        shapes_grid.orientation = Gtk.Orientation.VERTICAL;
+        shapes_grid.width_request = 200;
+        shapes_grid.name = "shapes-menu";
+
+        var back_button = new Gtk.ModelButton ();
+        back_button.text = _("Add Items");
+        back_button.inverted = true;
+        back_button.menu_name = "main";
+
+        var sub_separator = new Gtk.Separator (Gtk.Orientation.HORIZONTAL);
+        sub_separator.margin_top = sub_separator.margin_bottom = 3;
+
+        var rectangle = new Akira.Partials.PopoverButton (_("Rectangle"), "shape-rectangle-symbolic", {"R"});
+        rectangle.action_name = Akira.Services.ActionManager.ACTION_PREFIX +
+            Akira.Services.ActionManager.ACTION_ADD_RECT;
+
+        var ellipse = new Akira.Partials.PopoverButton (_("Ellipse"), "shape-circle-symbolic", {"E"});
+        ellipse.action_name = Akira.Services.ActionManager.ACTION_PREFIX +
+            Akira.Services.ActionManager.ACTION_ADD_ELLIPSE;
+
+        shapes_grid.add (back_button);
+        shapes_grid.add (sub_separator);
+        shapes_grid.add (rectangle);
+        shapes_grid.add (ellipse);
+        shapes_grid.show_all ();
+
+        var shapes_button = new Gtk.ModelButton ();
+        shapes_button.text = _("Shapes");
+        shapes_button.menu_name = "shapes-menu";
+
+        var separator2 = new Gtk.Separator (Gtk.Orientation.HORIZONTAL);
+        separator2.margin_top = separator2.margin_bottom = 3;
+
+        var vector = new Akira.Partials.PopoverButton (_("Vector"), "segment-curve", {"V"});
+
+        var pencil = new Akira.Partials.PopoverButton (_("Pencil"), "edit-symbolic", {"P"});
+
+        var text = new Akira.Partials.PopoverButton (_("Text"), "shape-text-symbolic", {"T"});
+        text.action_name = Akira.Services.ActionManager.ACTION_PREFIX +
+            Akira.Services.ActionManager.ACTION_ADD_TEXT;
+
+        var image = new Akira.Partials.PopoverButton (_("Image"), "image-x-generic-symbolic", {"I"});
+
+        grid.add (artboard);
+        grid.add (separator);
+        grid.add (shapes_button);
+        grid.add (separator2);
+        grid.add (vector);
+        grid.add (pencil);
+        grid.add (text);
+        grid.add (image);
+        grid.show_all ();
+
+        var popover = new Gtk.PopoverMenu ();
+        popover.add (grid);
+        popover.add (shapes_grid);
+        popover.child_set_property (grid, "submenu", "main");
```

Comment:
Can we rename main to main-menu here?

Reply:
I did that at the beginning but I'm getting a Gtk Warning saying the "main" submenu can't be found in the Gtk.Stack.
I wonder if PopoverMenu expects the main menu to be always called "main"

Reply:
Can you check using Inspector the name of each child, maybe something is missing somewhere?

Reply:
![Screenshot from 2019-05-06 21-49-51](https://user-images.githubusercontent.com/2527103/57272569-1b08dd00-7049-11e9-97a8-5bfaad129cc4.png)

Reply:
I don't think this is a big deal as everything works as expected and the menu name is not visible to the user but only used by GTK. What do you think?

Reply:
Yeah it was just a nitpick. Let's merge this too.